### DISCUSSION TITLE
Why are optimization results from cartesian products not reproducible by default?

### DISCUSSION THREAD

**Discussion by binarymason** [kernc/backtesting.py#302] [Q&A]
If optimization is doing an exhaustive search through all of the combinations of the cartesian product of parameters, why do I need to set `random_state` to get reproducible results? I would think an exhaustive search would give me reproducible results no matter what. 

Thanks!

https://github.com/kernc/backtesting.py/blob/e3cccdfc0fdf8f6f1fe0b5c7b84fd5a2b15fab2d/backtesting/backtesting.py#L1254

**Comment by kernc** [comment]
`random_state=` comes in use when `max_tries=` is set.
https://github.com/kernc/backtesting.py/blob/e3cccdfc0fdf8f6f1fe0b5c7b84fd5a2b15fab2d/backtesting/backtesting.py#L1216-L1222
